### DISCUSSION TITLE
During a time step, what order do agents act in?

### DISCUSSION THREAD

**Discussion by fl-hat** [JoelForamitti/agentpy#26] [Q&A] [answered]
Hi! I was wondering if multiple agents move during the same time step, and their positions are affected by the positions of other agents, how does agentpy determine who moves first? Is it randomized, sequential by agent ID, or something else? 

For example, in the segregation model, at each time step unhappy agents are directed to `find_new_home(self)`. In that method, they're randomly positioned at an empty spot (`self.model.grid.empty`). If all the agents started off as unhappy, who would be the first agent to move to a new spot during the first time step? Does the first agent to move change every time step?

Thank you so much!

**Comment by jofmi** [comment, marked as answer]
You can define the order of actions manually. Assuming that `agents` is an `AgentList` with your agents, you can

- call them in order of the list with `agents.my_action()`
- in a specific order using `agents.sort(...).my_action()`
- in a random order using `agents.shuffle().my_action()`

You can also split actions in decision and effect and call the first for every agent before calling the second:

```
agents.my_decision()
agents.my_effect()
```

In some models, this can be used to represent simultaneous actions.
